src/routes/settings/OpenAI.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+	import { MonitorUp } from 'lucide-svelte';
+	import { toast } from 'svelte-sonner';
+
+	import LL from '$i18n/i18n-svelte';
+	import { OpenAIStrategy } from '$lib/chat/openai';
+	import Button from '$lib/components/Button.svelte';
+	import FieldHelp from '$lib/components/FieldHelp.svelte';
+	import FieldInput from '$lib/components/FieldInput.svelte';
+	import Fieldset from '$lib/components/Fieldset.svelte';
+	import P from '$lib/components/P.svelte';
+	import { settingsStore } from '$lib/localStorage';
+
+	const DEFAULT_OPENAI_SERVER = 'https://api.openai.com/v1';
+
+	let openai = new OpenAIStrategy();
+	let openaiServer = $settingsStore.openaiServer || DEFAULT_OPENAI_SERVER;
+	let openaiApiKey = $settingsStore.openaiApiKey || '';
+	let openaiServerStatus: 'connected' | 'disconnected' | 'connecting' = 'disconnected';
+
+	$: settingsStore.update((settings) => ({ ...settings, openaiServer, openaiApiKey }));
+
+	async function getModelsList(): Promise<void> {
+		openaiServerStatus = 'connecting';
+		toast.warning($LL.connecting());
```

Comment:
Let's use the "loading" toast instead of warning, [here's an example](https://github.com/fmaclen/hollama/blob/main/src/routes/settings/Ollama.svelte#L53).